### PR TITLE
fix: studio reloads with updated spec on save even when watchForFileChanges is false

### LINKED ISSUE #14744: Setting watchForFileChanges to false keeps the Studio open
### Current behavior

Steps to replicate:
1. set `watchForFileChanges` to `false` in `cypress.json`
2. open a test and start a studio
3. interact with your app and click on "save commands"
4. nothing happens, studio remains open, however, scneario is added to my test file - multiple times if I click the button multiple times. which I do, because it feels unresponsive 🙈

### Desired behavior

When the `watchForFileChanges` is set to `true`, clicking on „save commands“ will close the studio. I believe this would be better UX for this case as well

### Test code to reproduce

I explored this in my stream, you can [see the behavior at 16:20](https://youtu.be/f_GAyKdD8OE?t=981)

### Versions
Cypress 6.3.1

### PR DESCRIPTION
**Pull request by panzarino**
- Closes #14744

### User facing changelog
Fixes an issue where studio would not exit with configuration `"watchForFileChanges": false`

### Additional details
So I found it pretty difficult to write any tests for this that were effective so there are no tests included in this fix. I'd be open to adding unit/integration tests for `event-manager.js` but not sure what would be possible outside of that.

### How has the user experience changed?
Studio will always exit and the TR will rerun all tests regardless of what `watchForFileChanges` is set to

### PR Tasks

- [ ] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Review by fornever847 — DISMISSED** [review]

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
@panzarino It seems the tests are failing - or the snapshots need updating. 

So...it does save the new commands to the file upon 'save', but after hitting 'save' and tests rerunning, the new commands are not run. I have to refresh the whole browser to see new commands. I guess this....kind of makes sense when you have watchForFileChanges off? But, seems unexpected - like a bug because the whole test suite 'reran' and it's not there.

**Comment by panzarino** [comment]
@jennifer-shehane snapshots need updating - super weird effect of how the studio tests work

For the reloading - I tested super quickly and didn't notice that it wasn't actually loading the tests, going to update

**Comment by panzarino** [comment]
@jennifer-shehane I've updated it so that it'll temporarily watch the file when saved and reload with the updated tests appropriately

File: packages/server/lib/socket.js [hunk 1]
```diff
@@ -71,7 +73,17 @@ class Socket {
     })
   }
 
-  watchTestFileByPath (config, specConfig, options) {
+  onStudioTestFileChange (filePath) {
+    return this.onTestFileChange(filePath).then(() => {
```
Comment (flotwig) [inline packages/server/lib/socket.js:77]: may be worth adding a comment? was non-obvious to me what this was doing
```suggestion
    // wait for the studio test file to be written to disk, then reload the test
    return this.onTestFileChange(filePath).then(() => {
```

**Review by flotwig — CHANGES_REQUESTED** [review]

File: packages/server/lib/socket.js (hunk 1 above)
Comment (panzarino) [reply packages/server/lib/socket.js:79]: I've added some additional comments to clarify things

**Review by flotwig — APPROVED** [review]

**Pull request merged**